packages/cli/generators/discover/index.js

```diff
@@ -0,0 +1,292 @@
+    const prompts = [
+      {
+        name: 'dataSource',
+        message: `Select the connector to discover`,
+        type: 'list',
+        choices: this.dataSourceChoices,
+        when: this.artifactInfo.dataSource === undefined,
+      },
+    ];
+
+    return this.prompt(prompts).then(answer => {
+      if (!answer.dataSource) return;
+      debug(`Datasource answer: ${JSON.stringify(answer)}`);
+
+      this.artifactInfo.dataSource = this.dataSourceChoices.find(
+        d => d.name === answer.dataSource,
+      );
+    });
+  }
+
+  /**
+   * Puts all discoverable models in this.modelChoices
+   */
+  async discoverModelInfos() {
+    if (this.artifactInfo.modelDefinitions) return;
+    debug(`Getting all models from ${this.artifactInfo.dataSource.name}`);
+    this.modelChoices = await modelMaker.discoverModelNames(
+      this.artifactInfo.dataSource,
+      {views: this.options.views, schema: this.options.schema},
+    );
+    debug(
+      `Got ${this.modelChoices.length} models from ${
+        this.artifactInfo.dataSource.name
+      }`,
+    );
+  }
+
+  /**
+   * Now that we have a list of all models for a datasource,
+   * ask which models to discover
+   */
+  promptModelChoices() {
+    // If we are discovering all we don't need to prompt
+    if (this.options.discoverAll) {
+      this.discoveringModels = this.modelChoices;
+    }
+
+    const prompts = [
+      {
+        name: 'discoveringModels',
+        message: `Select the models which to discover`,
+        type: 'checkbox',
+        choices: this.modelChoices,
+        when: this.discoveringModels === undefined,
+      },
+    ];
+
+    return this.prompt(prompts).then(answers => {
+      if (!answers.discoveringModels) return;
+      debug(`Models chosen: ${JSON.stringify(answers)}`);
+      this.discoveringModels = [];
+      answers.discoveringModels.forEach(m => {
+        this.discoveringModels.push(this.modelChoices.find(c => c.name === m));
+      });
+    });
+  }
+
+  /**
+   * Using artifactInfo.dataSource,
+   * artifactInfo.modelNameOptions
+   * and artifactInfo.modelDefinitionOptions
+   *
+   * this will discover every model
+   * and put it in artifactInfo.modelDefinitions
+   * @return {Promise<void>}
+   */
+  async getAllModelDefs() {
+    if (this.artifactInfo.modelDefinitions) return;
+    this.artifactInfo.modelDefinitions = [];
+    for (let i = 0; i < this.discoveringModels.length; i++) {
+      const modelInfo = this.discoveringModels[i];
+      debug(`Discovering: ${modelInfo.name}...`);
+      this.artifactInfo.modelDefinitions.push(
+        await modelMaker.discoverSingleModel(
+          this.artifactInfo.dataSource,
+          this.artifactInfo.modelDefinitionOptions,
+          modelInfo.name,
+        ),
+      );
+      debug(`Discovered: ${modelInfo.name}`);
+    }
+  }
+
+  promptController() {
+    if (this.shouldExit()) return;
+    const prompts = [
+      {
+        name: 'generateRepository',
```

Comment:
The name should be `generateController`?

Reply:
Where are these answers being used? I don't see code to generate controller/repository classes based on this prompt.

Reply:
Fixed!  I had briefly toyed with entire api generation.